examples/Model/User.php

```diff
@@ -55,20 +55,11 @@ function apply(array $state, Message ...$events): array
     foreach ($events as $event) {
         switch ($event->messageName()) {
             case UserWasRegistered::class:
-                $state['version'] = 1;
-
-                return array_merge($state, $event->payload(), ['activated' => true]);
+                return array_merge($state, $event->payload(), ['activated' => true, 'version' => 1]);
             case UserWasRegisteredWithDuplicateEmail::class:
-                $state = array_merge($state, $event->payload(), ['activated' => false, 'blocked_reason' => 'duplicate email']);
-                ++$state['version'];
-
-                return $state;
+                return array_merge($state, $event->payload(), ['activated' => false, 'blocked_reason' => 'duplicate email', 'version' => ++$state['version']]);
```

Comment:
`++$state['version']` That is a problem, because you can use the apply method only once for the same events. When you replay and `state` already has a version set, this will increment the version which is wrong. Replay again and version is incremented again. This is a side-effect and very dangerous.